mapper/orm-coordination-database-polling/src/main/java/org/hibernate/search/mapper/orm/coordination/databasepolling/event/impl/DefaultOutboxEventFinder.java

```diff
@@ -23,17 +24,23 @@ public DefaultOutboxEventFinder create(Optional<OutboxEventPredicate> predicate)
 	}
 
 	public static String createQueryString(Optional<OutboxEventPredicate> predicate) {
-		return "select e from OutboxEvent e"
-				+ predicate.map( p -> " where " + p.queryPart( "e" ) ).orElse( "" )
-				+ " order by e.id";
+		ProcessAfterFilter processAfterFilter = new ProcessAfterFilter();
+		OutboxEventPredicate combined = ( predicate.isPresent() ) ?
+				OutboxEventAndPredicate.of( predicate.get(), processAfterFilter ) :
+				processAfterFilter;
+
+		return "select e from OutboxEvent e where " + combined.queryPart( "e" ) + " order by e.id";
 	}
 
 	public static Query<OutboxEvent> createQuery(Session session, int maxResults,
 			String queryString, Map<String, Object> params) {
 		Query<OutboxEvent> query = session.createQuery( queryString, OutboxEvent.class );
+
 		for ( Map.Entry<String, Object> entry : params.entrySet() ) {
 			query.setParameter( entry.getKey(), entry.getValue() );
 		}
+		query.setParameter( "now", Instant.now() );
```

Comment:
This is a bit of a hack.

Couldn't you rather change the OutboxEventPredicate interface to allow predicates to set their parameters dynamically?

E.g.:

```
public interface OutboxEventPredicate {
	String queryPart(String eventAlias);

	void setParams(Query<OutboxEvent> query);
}
```

And change `org.hibernate.search.mapper.orm.coordination.databasepolling.event.impl.DefaultOutboxEventFinder#createQuery` to accept an `Optional<OutboxEventPredicate> predicate` instead of a `Map<String, Object> params`, so that it calls `OutboxEventPredicate#setParams` each time we create a query.
The query string can still be generated ahead of time, that's not a problem.